### PR TITLE
React plotly wrapper

### PR DESCRIPTION
**Pull request by alisman**
Create a wrapper for plotly that handles tooltip using React and yet does not redraw chart on every chart hover and unhover event

Changes proposed in this pull request:
- a
- b

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: src/shared/components/reactPlotlyWrapper/ReactPlotlyWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+export interface IReactPlotlyWrapperProps {
+    layout:any;
+    data:any; buildTooltip:(tooltipModel:any)=>JSX.Element
+}
+
+@observer
+export default class ReactPlotlyWrapper extends React.Component<IReactPlotlyWrapperProps, {}>
+{
+
+    private wrapper:HTMLDivElement;
```
Comment (adamabeshouse) [inline src/shared/components/reactPlotlyWrapper/ReactPlotlyWrapper.tsx]: based on use this seems like it should be typed as a jquery object

File: src/shared/components/reactPlotlyWrapper/ReactPlotlyWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        }
+    }
+
+    renderPlotly(){
+
+        Plotly.newPlot(this.wrapper, this.props.data, this.props.layout);
+
+        const that = this;
+
+        this.wrapper.on('plotly_hover', function(data: any){
```
Comment (adamabeshouse) [inline src/shared/components/reactPlotlyWrapper/ReactPlotlyWrapper.tsx]: why not using arrow functions?

Comment (adamabeshouse) [reply src/shared/components/reactPlotlyWrapper/ReactPlotlyWrapper.tsx]: nevermind you need the context

File: src/shared/components/reactPlotlyWrapper/ReactPlotlyWrapper.tsx (hunk 1 above)
Comment (ersinciftci) [reply src/shared/components/reactPlotlyWrapper/ReactPlotlyWrapper.tsx]: Mine works like this too, no need to use jQuery.

Comment (adamabeshouse) [reply src/shared/components/reactPlotlyWrapper/ReactPlotlyWrapper.tsx]: further down we're using `wrapper.on`, on is a jQuery function isnt it

**Pull request merged**